src/pages/index.js

```diff
@@ -54,43 +64,213 @@ const addFormValidator = new FormValidator(
   addCardFormElement
 );
 
-section.renderItems();
-profileEditPopup.setEventListeners();
-addCardPopup.setEventListeners();
+avatarFormValidator.enableValidation();
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
-imagePopup.setEventListeners();
+
+function handleDeleteClick(cardId) {
+  if (!cardId || typeof cardId !== "string") {
+    console.error("Invalid cardId for deletion:", cardId);
+    return;
+  }
+  deleteCardPopup.open({ cardId: cardId });
+}
 
 function handleImageClick(data) {
   imagePopup.open(data);
 }
 
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const sanitizedLink =
+    cardData.link.replace(/["'\s]+$/, "").trim() ||
+    "https://placehold.co/150x150";
+  const card = new Card(
+    {
+      name: cardData.name,
+      link: sanitizedLink,
+      _id: cardData._id || null,
+      isLiked: cardData.isLiked !== undefined ? cardData.isLiked : false,
+    },
+    "#card-template",
+    api,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
+
   const element = card.getView();
   return element;
 }
 
 function handleProfileFormSubmit(data) {
-  userInfo.setUserInfo({
-    name: data["edit-profile"],
-    job: data["profile-description"],
-  });
-  profileEditPopup.close();
+  profileEditSaveButton.textContent = "Saving...";
+  profileEditSaveButton.classList.add("loading");
+  profileEditSaveButton.disabled = true;
+  api
+    .updateUserInfo(data["edit-profile"], data["profile-description"])
+    .then((updatedInfo) => {
+      userInfo.setUserInfo({
+        name: updatedInfo.name,
+        about: updatedInfo.about,
+        avatar: updatedInfo.avatar,
+      });
+      profileEditPopup.close();
+      profileEditSaveButton.textContent = "Save";
+      profileEditSaveButton.classList.remove("loading");
+      profileEditSaveButton.disabled = false;
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.